lib/spack/spack/cmd/dev_build.py

```diff
@@ -102,29 +103,40 @@ def dev_build(self, args):
     if not spack.repo.PATH.exists(spec.name):
         raise spack.repo.UnknownPackageError(spec.name)
 
-    if not spec.versions.concrete_range_as_version:
-        tty.die(
-            "spack dev-build spec must have a single, concrete version. "
-            "Did you forget a package version number?"
-        )
-
-    source_path = args.source_path
-    if source_path is None:
-        source_path = os.getcwd()
-    source_path = os.path.abspath(source_path)
-
-    # Forces the build to run out of the source directory.
-    spec.constrain("dev_path=%s" % source_path)
-    spec.concretize()
-
-    if spec.installed:
-        tty.error("Already installed in %s" % spec.prefix)
-        tty.msg("Uninstall or try adding a version suffix for this dev build.")
-        sys.exit(1)
-
-    # disable checksumming if requested
-    if args.no_checksum:
-        spack.config.set("config:checksum", False, scope="command_line")
+    env = ev.active_environment()
+    if env:
+        matches = env.all_matching_specs(spec)
+        dev_matches = [m for m in matches if m.is_develop]
+        if len(dev_matches) > 1:
+            tty.die("Too many matching develop specs in the active environment")
+        elif len(dev_matches) < 1:
+            tty.die("No matching develop specs found in the active environment")
+        else:
+            spec = dev_matches[0]
+    else:
+        if not spec.versions.concrete_range_as_version:
+            tty.die(
+                "spack dev-build spec must have a single, concrete version. "
+                "Did you forget a package version number?"
+            )
+
+        source_path = args.source_path
+        if source_path is None:
+            source_path = os.getcwd()
+        source_path = os.path.abspath(source_path)
+
+        # Forces the build to run out of the source directory.
+        spec.constrain("dev_path=%s" % source_path)
+        spec.concretize()
+
+        if spec.installed:
+            tty.error("Already installed in %s" % spec.prefix)
+            tty.msg("Uninstall or try adding a version suffix for this dev build.")
+            sys.exit(1)
+
+        # disable checksumming if requested
+        if args.no_checksum:
+            spack.config.set("config:checksum", False, scope="command_line")
```

Comment:
This can be deleted generally, right?

Reply:
There's no checksum arg action (though maybe there should be), so currently commands have to do this.

Reply:
point is no fetching happens because dev_path is set

Reply:
I believe it may fetch patches.

Reply:
Okay, then it should not be indented. Move to before the branch?

Reply:
Hmm missed all the discussion. Didn't update browser. This DOES fetching.  It works without an environment and basically let's you develop a single root spec. So these fetch commands also apply to the dependencies IIRC.  I've just been trying to get it to work with environments.